Missing line number for `yield from` an invalid type

**Bug Report**

Error for `yield from` from a malformed type that isn't `Iterable` is missing line number. This prevents me even from suppressing the error inline.

I wasn't able to reproduce it with simpler "bad" types. Direct attribute access works as intended.

**To Reproduce**

[playground](https://mypy-play.net/?mypy=master&python=3.12&flags=strict&gist=1db8eaa9af876e0b44cc149703a852b3)

```python
from collections.abc import Iterator

class A:
    def list(self) -> None: ...

    def foo(self) -> list[int]:
        return []

def fn() -> Iterator[int]:
    A().foo().sort()
    yield from A().foo()
```

**Expected Behavior**

`attr-defined` error correctly attached to `yield from` line.

**Actual Behavior**

```
main.py: error: "list?[builtins.int]" has no attribute "__iter__" (not iterable)  [attr-defined]
main.py:6: error: Function "__main__.A.list" is not valid as a type  [valid-type]
main.py:6: note: Perhaps you need "Callable[...]" or a callback protocol?
main.py:10: error: "list?[builtins.int]" has no attribute "sort"  [attr-defined]
```

**Your Environment**

- Mypy version used: 1.14.1, master (#58de753b6b76227ff726e9ce1888cfc24c7ba44b)
- Mypy command-line flags: both `--strict` and without it
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.12

--- a/mypy/checkexpr.py
+++ b/mypy/checkexpr.py
@@ -6141,6 +6141,7 @@ class ExpressionChecker(ExpressionVisitor[Type]):
             generic_generator_type = self.chk.named_generic_type(
                 "typing.Generator", [any_type, any_type, any_type]
             )
+            generic_generator_type.set_line(e)
             iter_type, _ = self.check_method_call_by_name(
                 "__iter__", subexpr_type, [], [], context=generic_generator_type
             )

--- a/test-data/unit/check-statements.test
+++ b/test-data/unit/check-statements.test
@@ -2289,11 +2289,12 @@ from collections.abc import Iterator
 class A:
     def list(self) -> None: ...
 
-    def foo(self) -> list[int]:
+    def foo(self) -> list[int]:  # E: Function "__main__.A.list" is not valid as a type \
+                                 # N: Perhaps you need "Callable[...]" or a callback protocol?
         return []
 
 def fn() -> Iterator[int]:
-    yield from A().foo()
+    yield from A().foo()  # E: "list?[builtins.int]" has no attribute "__iter__" (not iterable)
 [builtins fixtures/tuple.pyi]
 
 [case testNoCrashOnStarRightHandSide]

--- a/test-data/unit/check-statements.test
+++ b/test-data/unit/check-statements.test
@@ -2283,6 +2283,19 @@ def get_strings(foo: bool) -> Iterator[str]:
         yield "bar2"
 [builtins fixtures/tuple.pyi]
 
+[case testYieldFromInvalidType]
+from collections.abc import Iterator
+
+class A:
+    def list(self) -> None: ...
+
+    def foo(self) -> list[int]:
+        return []
+
+def fn() -> Iterator[int]:
+    yield from A().foo()
+[builtins fixtures/tuple.pyi]
+
 [case testNoCrashOnStarRightHandSide]
 x = *(1, 2, 3)  # E: can't use starred expression here
 [builtins fixtures/tuple.pyi]

